### PR TITLE
[FIX] Fill xml-id cache for many2many fields.

### PR DESCRIPTION
**Pull request by StefanRijnhart**

### CODE REVIEW

File: openerp/addons/test_impex/tests/test_export.py [hunk 1]
```diff
@@ -566,6 +566,9 @@ def test_multiple_records(self):
                 [u'', u'13'],
             ])
 
+    def test_xml_ids(self):
+        self.export(self.commands, fields=['value/id'])
```
Comment (StefanRijnhart) [inline openerp/addons/test_impex/tests/test_export.py:570]: While this test reproduces the problem, we don't test this module in our CI and other tests fail on our default language.

File: openerp/models.py [hunk 2]
```diff
@@ -855,10 +855,9 @@ def __export_xml_id(self):
             })
             return '__export__.' + name
 
-    def build_xml_id_cache(self, records_cache, xml_id_cache):
```
Comment (StefanRijnhart) [inline openerp/models.py:858]: I now pass the old `records_cache` argument as `self`.

File: openerp/models.py [hunk 3]
```diff
@@ -955,40 +954,43 @@ def get_values_clause_bulk(lst):
...
-            xml_id_cache[first_item._name][res_id] = '__export__.' + name
+            xml_id_cache[self._name][res_id] = '__export__.' + name
 
         # Do a bulk insert in the database for new records
         if to_create:
             clause, values = get_values_clause_bulk(to_create)
             self.env.cr.execute("INSERT INTO ir_model_data %s" % (clause), values)
 
     @api.multi
-    def __export_rows(self, fields, all_records=False):
```
Comment (StefanRijnhart) [inline openerp/models.py:972]: I now pass xml_id_cache recursively so that I don't start with an empty cache when exporting relational fields.

**Review by lembregtse — APPROVED** [review]
Appro

**Review by PieterPaulussen — APPROVED** [review]

**Pull request merged**